packages/labs/analyzer/src/lib/custom-elements/attributes.ts

```diff
@@ -0,0 +1,34 @@
+/**
+ * @fileoverview
+ *
+ * Utilities for analyzing attributes
+ */
+
+import ts from 'typescript';
+import {parseNamedTypedJSDocInfo} from '../javascript/jsdoc.js';
+import {Attribute} from '../model.js';
+import {AnalyzerInterface} from '../model.js';
+import {getTypeForTypeString} from '../types.js';
+
+/**
+ * Returns an array of analyzer `Event` models for the given
+ * ts.ClassDeclaration.
+ */
+export const addJSDocAttributeToMap = (
+  tag: ts.JSDocTag,
+  attributes: Map<string, Attribute>,
+  analyzer: AnalyzerInterface
+) => {
+  const info = parseNamedTypedJSDocInfo(tag, analyzer);
+  if (info === undefined) {
+    return;
+  }
+  const {name, type, description, summary} = info;
+  attributes.set(name, {
```

Comment:
what about merging, since we potentially have several ways to describe the same attribute? Do we have a good precedence defined if something overrides the other, or should we be careful to merge?

I think it could be a valid pattern to say use jsdoc to add attribute-only docs to an attribute otherwise defined by a reactive property, for instance.

We could drop a warning and todo if the map already contains the attribute for now.

Reply:
do we really need to warn? if i set up a converter, the attribute might be genuinely different from the field

Reply:
a contrived example of my previous comment, 

```js
@property({ converter: {...} }) constructors = [];
```
```html
<x-l constructors="a,b ,   c"></x-l>
<script>
console.log(document.currentScript.previousElementSibling.constructors);
// [object A], [object B], [object C]
</script>